Add mixin interfaces for shared attributes and events

heruan opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-text-field-flow/48)
<!-- Reviewable:end -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/vaadin-text-field-flow?pullRequest=48) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/vaadin-text-field-flow?pullRequest=48) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vaadin/vaadin-text-field-flow?pullRequest=48) it.</sub>

gilberto-torrezan commented:
Reviewed 8 of 8 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/vaadin-text-field-flow/48#-L4oWe2c2SFK7_CaK95r:-L4oWe2c2SFK7_CaK95s:bnprkle) (no related file):*
Thanks for the contribution @heruan ! I'll lock this PR for now - these new mixins could be part of Flow core (instead of just the `vaadin-text-field-flow` project), and they will need tests as well. We will consider adding those in the near future.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-text-field-flow/48)*
<!-- Sent from Reviewable.io -->

heruan commented:
Review status: 5 of 8 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/vaadin-text-field-flow/48#-L4oWe2c2SFK7_CaK95r:-L4oZslMq9hVeIP79xjb:b-bxq73j) (no related file):*
<details><summary><i>Previously, gilberto-torrezan (Gilberto Torrezan) wrote…</i></summary><blockquote>

Thanks for the contribution @heruan ! I'll lock this PR for now - these new mixins could be part of Flow core (instead of just the `vaadin-text-field-flow` project), and they will need tests as well. We will consider adding those in the near future.
</blockquote></details>

Great! I've just rebased on master, I can move the mixins in a PR on Flow core if you're okay with that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-text-field-flow/48)*
<!-- Sent from Reviewable.io -->

heruan commented:
Now this depends on https://github.com/vaadin/flow/pull/3517

heruan commented:
A bit of a recap from this: this was formerly to add `HasCompletion`, `HasCorrection`, `HasCapitalization` mixins which are currently being moved to Flow with vaadin/flow#3517; now, this is just the addition of the following of mixins to the components: `HasCompletion`, `HasCorrection`, `HasCapitalization`, `InputNotifier`, `KeyNotifier`, `CompositionNotifier`.

heruan commented:
@gilberto-torrezan I moved the mixins to Flow core as you suggested in https://github.com/vaadin/flow/pull/3517, then there @pleku considered they might better be part of the components since there are no other component in Flow which uses them.

So I moved them back to this repo in #66 and changed this with just the key notifiers mixins (which comes from Flow).

pleku commented:
Reviewed 3 of 8 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/vaadin/vaadin-text-field-flow/48)*
<!-- Sent from Reviewable.io -->

pleku commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/vaadin/vaadin-text-field-flow/48#-:-L9EB4q-6-93sxQAARKH:bnfp4nl)*
<!-- Sent from Reviewable.io -->
